torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1867,6 +1867,29 @@ def sample_inputs_std_var(op_info, device, dtype, requires_grad, **kwargs):
         SampleInput(tensor_1d, kwargs=dict(dim=0, unbiased=False, keepdim=False)),
     ]
 
+def sample_inputs_cov(op_info, device, dtype, requires_grad=False):
+    inp_dims = [ (S,), (1, S), (S, 2), (S, S)]  # (S, 1) produces nan outputs
+    biases = [True, False]
+    rowvars = [True, False]
+    ddofs = [None, S]
+    inputs = []
+    for inp, bias, rowvar, ddof in product(inp_dims, biases, rowvars, ddofs):
+        if np.size(inp) == 1:
+            t_size = inp[0]
+        elif rowvar or inp[0] == 1:
+            t_size = inp[1]
+        else:
+            t_size = inp[0]
+        fwt = torch.randint(1, 10, (t_size,), device=device) * 2 + 1  # '+1' to avoid zero sum weights
+        awt = torch.randint(1, 10, (t_size,), device=device)**2 + 1
+        tens = make_tensor(inp, device=device, dtype=dtype,
+                            low=-50, high=50, requires_grad=requires_grad)
+        tens2 = tens + make_tensor(inp, device=device, dtype=dtype,
+                            low=None, high=None, requires_grad=requires_grad)      
```

Comment:
@PandaBoi tens is part of tens2's graph, you should do something like `tens2 = tens2.detach()` so that there are no weird dependency issues between the inputs (and this might be the reason gradcheck is failing). If you'd like compute grad wrt to tens2 as well you'd need to do `tens2 = tens2.detach().requires_grad_(True)`

Reply:
In general though, since you are writing a composite op (where you don't have to explicitly compute the backward formula), there is not too much room for error in the implementation itself that might cause gradcheck to fail. Knowing this, you might try to pay more attention to what kinds of sample data you are feeding in.

Reply:
Awesome @soulitzer! That solved the problem right away.. Looks like it was tens2's mingling that was causing the issue.
Cheers!

Reply:
Thanks @soulitzer!